scene/gui/code_edit.cpp

```diff
@@ -1329,21 +1329,39 @@ bool CodeEdit::is_drawing_executing_lines_gutter() const {
 
 void CodeEdit::_main_gutter_draw_callback(int p_line, int p_gutter, const Rect2 &p_region) {
 	bool hovering = get_hovered_gutter() == Vector2i(main_gutter, p_line);
-	if (draw_breakpoints && theme_cache.breakpoint_icon.is_valid()) {
+	if (draw_breakpoints && theme_cache.breakpoint_icon.is_valid() && theme_cache.breakpoint_conditional_icon.is_valid() && theme_cache.breakpoint_no_suspend_icon.is_valid() && theme_cache.breakpoint_conditional_no_suspend_icon.is_valid()) {
 		bool breakpointed = is_line_breakpointed(p_line);
 		bool shift_pressed = Input::get_singleton()->is_key_pressed(Key::SHIFT);
 
 		if (breakpointed || (hovering && !is_dragging_cursor() && !shift_pressed)) {
 			int padding = p_region.size.x / 6;
 
 			Color use_color = theme_cache.breakpoint_color;
+			Dictionary dict = get_line_gutter_metadata(p_line, p_gutter);
+			if (!(bool)dict["bp_enabled"]) {
+				use_color = theme_cache.breakpoint_disabled_color;
+			} else if (!((String)dict["bp_print"]).is_empty() && !(bool)dict["bp_suspend"]) {
+				use_color = theme_cache.breakpoint_print_color;
+			}
```

Comment:
```suggestion
			if (breakpointed && !(bool)dict["bp_enabled"]) {
				use_color = theme_cache.breakpoint_disabled_color;
			} else if (breakpointed && !((String)dict["bp_print"]).is_empty() && !(bool)dict["bp_suspend"]) {
				use_color = theme_cache.breakpoint_print_color;
			}
```
Since these should only be set if there actually is a breakpoint there.